scripts/db_migrator.py

```diff
@@ -596,6 +596,61 @@ def migrate_route_table(self):
                     route_key = "ROUTE_TABLE:{}".format(route_prefix)
                 self.appDB.set(self.appDB.APPL_DB, route_key, 'weight','')
 
+
+    def update_edgezone_aggregator_config(self):
+        """
+        Update cable length configuration in ConfigDB for T0 neighbor interfaces
+        connected to EdgeZone Aggregator devices, while resetting the port values to trigger a buffer change
+        1. Find a list of all interfaces connected to an EdgeZone Aggregator device.
+        2. If all the cable lengths are the same, do nothing and return.
+        2. If there are different cable lengths, update CABLE_LENGTH values for these interfaces with a constant value of 40m.
+        4. Reset port values to trigger buffer manager event listener to bring across new buffer changes for specified ports
+        """
+        device_neighbor_metadata = self.configDB.get_table("DEVICE_NEIGHBOR_METADATA")
+        device_neighbors = self.configDB.get_table("DEVICE_NEIGHBOR")
+        cable_length = self.configDB.get_table("CABLE_LENGTH")
+        port_table = self.configDB.get_table("PORT")
+        edgezone_aggregator_devs = []
+        edgezone_aggregator_intfs = []
+        EDGEZONE_AGG_CABLE_LENGTH = "40m"
+        for k, v in device_neighbor_metadata.items():
+            if "type" in v:
+                if v["type"] == "EdgeZoneAggregator":
+                    edgezone_aggregator_devs.append(k)
+
+        if len(edgezone_aggregator_devs) == 0:
+            return
+
+        for intf, intf_info in device_neighbors.items():
+            if intf_info["name"] in edgezone_aggregator_devs:
+                edgezone_aggregator_intfs.append(intf)
+
+        cable_length_table = self.configDB.get_entry("CABLE_LENGTH", "AZURE")
+        curr_cable_intf = next(iter(cable_length_table))
+        curr_cable_length = cable_length_table[curr_cable_intf]
+        index = 0
+
+        for intf, length in cable_length_table.items():
+            index += 1
+            if curr_cable_length != length:
+                break
+            elif index == len(cable_length_table):
+                """ All cable lengths are the same, nothing to modify """
+                return
+
+        for intf, length in cable_length_table.items():
+            if intf in edgezone_aggregator_intfs:
+                cable_length_table[intf] = EDGEZONE_AGG_CABLE_LENGTH
+
+        """ Set new cable length values """
+        self.configDB.set_entry("CABLE_LENGTH", "AZURE", cable_length_table)
```

Comment:
Move this inside for loop in 641. use self.configDB.set(self.configDB.CONFIG_DB, "CABLE_LENGTH|AZURE", intf, EDGEZONE_AGG_CABLE_LENGTH)